cfgen/cfgen

```diff
@@ -813,8 +824,30 @@ def _infinite_counter(start: int = 0) -> Iterator[int]:
 fidk_gen = _infinite_counter()  # fid key generator
 
 
+def __allocate_base_fidkey(objt: ObjT, fidk: int) -> int:
+    """
+    1. Gets the mask for the key based on ObjT.
+    mask = 0xffffffff00000000
+    2. Inverts the mask to set the key in the previosly ON bits.
+    mask ^ 0xffffffffffffffff = 0xffffffff00000000
+    >>> fidk=5
+    >>> hex((mask ^ 0xffffffffffffffff) * fidk + fidk)
+    '0x500000000'
+    3. Reset the remaining 32 bits by ANDing with original mask.
+    >>> hex(result & mask)
+    '0x500000000'
+    """
+    obj_mask = ObjT_Mask_Map.get(objt)
+    if obj_mask:
+        fidk = (((obj_mask.key ^ 0xffffffffffffffff) * fidk + fidk) &
+                obj_mask.key)
+    return fidk
+
+
 def new_oid(objt: ObjT) -> Oid:
-    return Oid(objt, next(fidk_gen))
+    fidk = next(fidk_gen)
```

Comment:
We need to save the last generated fidk in consul, it would be good to have it as a part of consul_kv.json.

Reply:
Done, .. kept the key like `last_base_fid_key/process`, `last_base_fid_key/sdev`, etc. 
We could retrieve keys with same prefix.. `last_base_fid_key/*`

For  `"last_dynamic_fid_key/process" : 0`, kept 0 for all objects.
Let me know if I need to change anything.